torch/_export/graph_module.py

```diff
@@ -224,59 +194,88 @@ def op_to_str(op):
         return (reduce_graph_module, (pickled_state,))
 
 
-def attach_export_graph_metadata(gm: fx.GraphModule, meta: ExportMetadata) -> None:
-    gm.meta[EXPORT_METADATA] = meta
-    gm.__class__ = type(type(gm).__name__, (ExportGraphModuleMixin, type(gm)), {})
+class ExportedProgram:
+    def __init__(
+        self,
+        root: Union[torch.nn.Module, Dict[str, Any]],
+        graph: torch.fx.Graph,
+        graph_signature: ExportGraphSignature,
+        call_spec: CallSpec,
+        state_dict: Dict[str, Any],
+    ):
+        # Remove codegen related things from the graph. It should just be a flat graph.
+        graph._codegen = torch.fx.graph.CodeGen()
+        gm = torch.fx.GraphModule(root, graph, class_name="ExportGraphModule")
+        gm.__class__ = type(type(gm).__name__, (ExportGraphModuleMixin, type(gm)), {})
```

Comment:
wow what's this?

Reply:
Hahah it's to have the graph module subclass both fx.GraphModule and ExportGraphModuleMixin so that we can override its pickling... all creds to Han...idk how it works.

Reply:
Do we need to keep pickling or are we going to eliminate it soon? Because serialization to Python dataclass should eliminate the need for it, right?

Reply:
Updated the code to remove the ExportGraphModuleMixin because we don't need to support pickling once serialization is added.